feature/navigation/drawer/src/main/kotlin/app/k9mail/feature/navigation/drawer/ui/folder/FolderList.kt

```diff
@@ -29,24 +33,79 @@ internal fun FolderList(
     val folderNameFormatter = remember { FolderNameFormatter(resources) }
     val listState = rememberLazyListState()
 
+    // Converting folders to TreeFolder
```

Comment:
The whole conversion to `TreeFolder` should be done using a use case, so it is easily testable and the `maxDepth` could be provided as attribute to the call. This needs to go through the view model and then propagated as state to the view. 

The FolderList should take a `TreeFolder` list and just render it.

Reply:
@wmontwe thanks a lot for this feedback, I tried something on eb97d6bc595d318b244967ad70be79562fc43a83, can you review it please ?